packages/composable-controller/tsconfig.json

```diff
@@ -6,6 +6,9 @@
   "references": [
     {
       "path": "../base-controller"
+    },
+    {
+      "path": "../json-rpc-engine"
     }
```

Comment:
Added `json-rpc-engine` as a TypeScript project reference (but not in the build tsconfig) instead of introducing it as a dependency, since it's only being used in a test file.

Reply:
Do we also want to add `json-rpc-engine` as a dev dependency? Maybe it makes no difference from a TypeScript perspective, but it communicates the dependency from a Yarn/NPM perspective.

Reply:
Fixed here: https://github.com/MetaMask/core/pull/3904/commits/5da089327847068f62dd1ec971b5a3299bff8eb9

Will a `json-rpc-engine` project reference need to be added to `tsconfig.build.json` as well?

Reply:
If it's only being used for tests, then you're correct, I don't think adding it to `tsconfig.build.json` would be needed. You should be able to test this by running `yarn build` — you shouldn't see any errors.

Reply:
No errors on yarn build! Think we're good to go.